kratos/python_scripts/KratosUnittest.py

```diff
@@ -160,10 +160,13 @@ class WorkFolderScope:
     def __init__(self, rel_path_work_folder, file_path):
         """file_path is the __file__ argument"""
         self.currentPath = os.getcwd()
+        self.currentPythonpath = sys.path
         self.scope = os.path.abspath(os.path.join(os.path.dirname(os.path.realpath(file_path)), rel_path_work_folder))
 
     def __enter__(self):
         os.chdir(self.scope)
+        sys.path.append(self.scope)
```

Comment:
changes to the core (esp such a class) affect everyone, please make a separate PR explaining why the changes are needed

btw I am not sure if this can go in, we currently try to remove adding things to the path bcs it messes things up
=> to be discussed in a separate PR

Reply:
I discussed with @roigcarlo and he proposed to put this change in this specific PR since it is needed in this specific PR.

Reply:
@philbucher As @riccardotosi says this is somehow hard to understand without an specific example which needs to use this functionality. Hance why is in here. but if you preffer we candiscuss it in a separate PR.

But as the topic has already appeared, a little explanation is in order:
The problem here is caused because the specific example we need to run assumes that, apart from the regular MainKratos.py, the directory from which the problem will run has an aditional *.py file. Unfortunatelly right now there is no other way to proced with this. 

Normally this would not be problem, but since the Unittest library runs the tests from an arbitrary palce we need to tell python where to look for this file. Since we already have a function which manipulates the dir it seemed logic to put it here.

Reply:
no worries @roigcarlo :)

Why don't you manually add the path of the test-file to the python-path in the respective test? This should work if the "other" py-file is in the same directory

I am just afraid if we add it by default then we could hide some import-problems :/

=> if my first proposal doesn't work then could you make the adding to the path an optional thing? I.e. a constructor argument defaulting to `False`?

Reply:
I like more the second option, maybe we can make a separate context manager just for paths as we have for dirs, I didn't think on that one honestly

Reply:
@roigcarlo up to you :) 
I think here a default arg could make more sense, since the rest of the contextmanager is the same (as far as I understood) 
=> less chances of the implementations to diverge in the future

@riccardotosi I will remove my block as soon as someone reviewed the rest, since it is currently not possible to dismiss reviews and I cannot approve this